cluster-autoscaler/processors/customresources/dra_processor_test.go

```diff
@@ -0,0 +1,187 @@
+	start := time.Now()
+	later := start.Add(10 * time.Minute)
+	expectedReadiness := make(map[string]bool)
+
+	readyCondition := apiv1.NodeCondition{
+		Type:               apiv1.NodeReady,
+		Status:             apiv1.ConditionTrue,
+		LastTransitionTime: metav1.NewTime(later),
+	}
+	unreadyCondition := apiv1.NodeCondition{
+		Type:               apiv1.NodeReady,
+		Status:             apiv1.ConditionFalse,
+		LastTransitionTime: metav1.NewTime(later),
+	}
+
+	node1DraReady := &apiv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "node_1_Dra_Ready",
+			CreationTimestamp: metav1.NewTime(start),
+		},
+		Status: apiv1.NodeStatus{
+			Capacity:    apiv1.ResourceList{},
+			Allocatable: apiv1.ResourceList{},
+			Conditions:  []apiv1.NodeCondition{readyCondition},
+		},
+	}
+	expectedReadiness[node1DraReady.Name] = true
+
+	node2DraReady := &apiv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "node_2_Dra_Ready",
+			CreationTimestamp: metav1.NewTime(start),
+		},
+		Status: apiv1.NodeStatus{
+			Capacity:    apiv1.ResourceList{},
+			Allocatable: apiv1.ResourceList{},
+			Conditions:  []apiv1.NodeCondition{readyCondition},
+		},
+	}
+	expectedReadiness[node2DraReady.Name] = true
+
+	node3DraUnready := &apiv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "node_3_Dra_Unready",
+			CreationTimestamp: metav1.NewTime(start),
+		},
+		Status: apiv1.NodeStatus{
+			Capacity:    apiv1.ResourceList{},
+			Allocatable: apiv1.ResourceList{},
+			Conditions:  []apiv1.NodeCondition{readyCondition},
+		},
+	}
+	expectedReadiness[node3DraUnready.Name] = false
+
+	nodeNoDraReady := &apiv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "node_4_NonDra_Ready",
+			CreationTimestamp: metav1.NewTime(start),
+		},
+		Status: apiv1.NodeStatus{
+			Conditions: []apiv1.NodeCondition{readyCondition},
+		},
+	}
+	expectedReadiness[nodeNoDraReady.Name] = true
+
+	nodeNoDraUnready := &apiv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "node_5_NonDra_Unready",
+			CreationTimestamp: metav1.NewTime(start),
+		},
+		Status: apiv1.NodeStatus{
+			Conditions: []apiv1.NodeCondition{unreadyCondition},
+		},
+	}
+	expectedReadiness[nodeNoDraUnready.Name] = false
+
+	initialReadyNodes := []*apiv1.Node{
+		node1DraReady,
+		node2DraReady,
+		node3DraUnready,
+		nodeNoDraReady,
+	}
+	initialAllNodes := []*apiv1.Node{
+		node1DraReady,
+		node2DraReady,
+		node3DraUnready,
+		nodeNoDraReady,
+		nodeNoDraUnready,
+	}
+
+	localSlices := map[string][]*resourceapi.ResourceSlice{
+		"node_1_Dra_Ready":   {node1Slice1, node1Slice2},
+		"node_2_Dra_Ready":   {node2Slice1, node2Slice2},
+		"node_3_Dra_Unready": {node3Slice1, node3Slice2},
+	}
+
+	provider := testprovider.NewTestCloudProviderBuilder().Build()
+	provider.SetMachineTemplates(map[string]*framework.NodeInfo{"m1": framework.NewNodeInfo(node1DraReady, []*resourceapi.ResourceSlice{node1Slice1, node1Slice2}), "m2": framework.NewTestNodeInfo(nodeNoDraReady)})
```

Comment:
I'd make sure that the template NodeInfo has different names etc. from the real Nodes, this is what happens in reality.